identity/registry/endpoints.go

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2017 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package registry
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/julienschmidt/httprouter"
+	"github.com/mysterium/node/tequilapi/utils"
+	"net/http"
+)
+
+// SignatureDTO represents Elliptic Curve signature parts
+//
+// swagger:model DecomposedSignatureDTO
+type SignatureDTO struct {
+	// S part of signature
+	// example: "0x1321313212312..."
+	R string
+	// R part of signature
+	// example: "0x1234563564354..."
+	S string
+	// Sign - 27 or 28 as expected by ethereum ecrecover function
+	// example: 27
+	V uint8
+}
+
+// PublicKeyPartsDTO represents ECDSA public key with first byte stripped (0x04) and splitted into two 32 bytes size arrays
+//
+// swagger:model PublicKeyPartsDTO
+type PublicKeyPartsDTO struct {
+	// First 32 bytes of public key in hex representation
+	// example: "0x1321313212312..."
+	Part1 string
+	// Last 32 bytes of public key inx hex representation
+	// example: "0x1321313212312..."
+	Part2 string
+}
+
+// RegistrationDataDTO represents registration status and needed data for registering of given identity
+//
+// swagger:model RegistrationDataDTO
+type RegistrationDataDTO struct {
+	// Returns true if identity is registered in payments smart contract
+	Registered bool
+
+	PublicKey *PublicKeyPartsDTO `json:"PublicKey,omitempty"`
```

Comment:
To keep same styling, let's start JSON values names from lower case:
PublicKey -> publicKey
Signature -> signature

Reply:
👍

Reply:
Improved